tests/test_enforce.py

```diff
@@ -95,3 +103,62 @@ def nonfinal_classmethod(cls):
 
         self.assertNotEqual(ClassMethodOverrider.nonfinal_classmethod(),
                             Enforcing.nonfinal_classmethod())
+
+    def test_is_compatible_when_compatible(self):
+        def foo(x, /, y: str, *args, z: int, **kwargs) -> object:
+            pass
+
+        def bar(x, /, y: object, a: float, *, b: bytes, z: int, **kwargs) -> str:
+            pass
+
+        is_compatible(foo, bar)
+
+    def test_is_compatible_when_return_types_are_incompatible(self):
+        def foo(x) -> int:
+            pass
+
+        def bar(x) -> str:
+            pass
+
+        with self.assertRaises(TypeError):
+            is_compatible(foo, bar)
+    
+    def test_is_compatible_when_parameter_positions_are_incompatible(self):
+        def foo(x, y):
+            pass
+
+        def bar(y, x):
+            pass
+
+        with self.assertRaises(TypeError):
+            is_compatible(foo, bar)
+    
+    def test_is_compatible_when_parameter_types_are_incompatible(self):
+        def foo(x: object):
+            pass
+
+        def bar(y: str):
+            pass
+
+        with self.assertRaises(TypeError):
+            is_compatible(foo, bar)
+
+    def test_is_compatible_when_parameter_kinds_are_incompatible(self):
+        def foo(x, *, y):
+            pass
+
+        def bar(x, y):
+            pass
+
+        with self.assertRaises(TypeError):
+            is_compatible(foo, bar)
```

Comment:
I think this should be allowed. `foo` is stricter and `bar` can replace it where ever `foo` is used. Subtype can have more loose type def.